src/envoy/http/authn/authn_utils.cc

```diff
@@ -48,6 +51,30 @@ void ExtractJwtAudience(
     // Not convertable to string array
   }
 }
+
+// Extract JWT groups into the JwtPayload.
+// This function should to be called after the claims are extracted.
+void ExtractJwtGroups(
+    const Envoy::Json::Object& obj,
+    const ::google::protobuf::Map< ::std::string, ::std::string>& claims,
+    istio::authn::JwtPayload* payload) {
+  const std::string& key = kJwtGroupsKey;
```

Comment:
Can we extend this to any string array claims?

Reply:
The claim key (e.g., kJwtGroupsKey, kJwtAudienceKey)  can be extended as an argument of the extraction function to support extracting any claims with the key. However, it is not straightforward to make the following statements in these extraction functions to be generic for extracting any claim key. If you have a way to do it, please let me know.
payload->add_audiences(claims.at(key));
payload->add_groups(claims.at(key));